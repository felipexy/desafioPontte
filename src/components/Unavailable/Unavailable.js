import React from 'react';
import './Unavailable.scss';

const Unavailable = () => {
  return (
    <div className="unavailable">
      <h1 data-testid="unaTest" className="unavailable__notWorking">
        Sistema fora do ar, tente novamente mais tarde! :(
      </h1>
    </div>
  );
};

export default Unavailable;
