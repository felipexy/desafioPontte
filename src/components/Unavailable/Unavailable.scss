.unavailable {
  display: flex;
  flex-direction: column;
  border-radius: 15px;
  position: relative;
  width: 90%;
  background: white;
  margin-right: auto;
  margin-left: auto;
  top: 100px;
}
    
.unavailable::after {
  content: "";
  display: block;
  padding-bottom: 2%;
}
  

.unavailable__notWorking{
  margin-right: auto;
  margin-left: auto;
}