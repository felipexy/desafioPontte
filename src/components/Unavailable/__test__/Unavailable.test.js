import React from 'react';
import Unavailable from '../Unavailable';
import { render, cleanup, waitFor, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

afterEach(cleanup);

describe('Tests for Unavailable component', () => {
    
  it('Renders right phrase', async () => {
    const { getByTestId } = render(<Unavailable />);
    const unvlbTst = await waitFor(() => getByTestId('unaTest'));
    expect(unvlbTst).toHaveTextContent('Sistema fora do ar, tente novamente mais tarde! :(');
  });

  it('Matches snapshot', () => {
    const tree = renderer.create(<Unavailable />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
