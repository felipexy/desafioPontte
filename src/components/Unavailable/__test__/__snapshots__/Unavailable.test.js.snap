// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Unavailable component Matches snapshot 1`] = `
<div
  className="unavailable"
>
  <h1
    className="unavailable__notWorking"
    data-testid="unaTest"
  >
    Sistema fora do ar, tente novamente mais tarde! :(
  </h1>
</div>
`;
